packages/editor/src/package/storage-format.ts

```diff
@@ -65,30 +69,41 @@ export function createEmptyDocument(
   }
 }
 
-export function migrate(stateBeforeMigration: unknown): {
+export function migrate(
+  stateBeforeMigration: unknown,
+  variant: EditorVariant
+): {
   migratedState: StorageFormat
   stateChanged: boolean
 } {
-  if (!t.type({ version: t.number }).is(stateBeforeMigration))
-    throw new Error(
-      `Missing property 'version: number' in state while trying to perform migrations. Got ${JSON.stringify(stateBeforeMigration)}`
-    )
-
-  // Only apply migrations for this version number and higher
-  const nextMigrationIndex = stateBeforeMigration.version
+  let migratedState: StorageFormat
+  let stateChanged = false
 
-  // Create deep copy
-  let migratedState = JSON.parse(JSON.stringify(stateBeforeMigration)) as {
-    version: number
+  // Check if the state is the (old format)
+  if (
+    DocumentType.is(stateBeforeMigration) &&
+    !StorageFormat.is(stateBeforeMigration)
+  ) {
+    migratedState = {
+      ...createEmptyDocument(variant),
```

Comment:
This would mix up the state types: On one hand we have all the properties of `version: 1` but on the other hand we set `version: 0`.